src/fmu/dataio/providers/_filedata.py

```diff
@@ -40,33 +43,30 @@ class FileDataProvider:
     rootpath: Path = field(default_factory=Path)
     itername: str = ""
     realname: str = ""
+    obj: Optional[types.Inferrable] = field(default=None)
+    compute_md5: bool = False
 
     # storing results in these variables
-    relative_path: Path = field(default_factory=Path)
-    relative_path_symlink: Optional[Path] = field(default=None)
-
-    absolute_path: Path = field(default_factory=Path)
-    absolute_path_symlink: Optional[Path] = field(default=None)
-
-    checksum_md5: Optional[str] = field(default="", init=False)
     forcefolder_is_absolute: bool = field(default=False, init=False)
 
     @property
     def name(self) -> str:
         return self.dataio.name or self.objdata.name
 
-    def derive_filedata(self) -> None:
+    def derive_filedata(self) -> meta.File:
```

Comment:
I very much like this change and think all providers should return the model similarly. We should even consider codifying this structure, whether lightly through a [Protocol](https://docs.python.org/3/library/typing.html#typing.Protocol) or a base class that requires all providers implement some common function (i.e. `derive_metadata()`?). 

To that end, maybe you could consider naming this method to something we could carry to all providers

Reply:
yes, completely agree, it will be very good to get common structure between them 🙂 I will change the method name  maybe to `get_metadata` or `get_metadata_model`?.. naming things are difficult 😄  suggestions are welcome!!
I guess we might change it again when designing the new structure anyway though..